### PR TITLE
[CDAP-15787] Updated ValueMapper for new Validation API

### PR DESCRIPTION
**Pull request by dli357**
Updated ValueMapper.java and ValueMapperTest.java, added additional validation to ensure that the mapped columns exist in the input and that the default mappings exist in the mapped columns. Added additional unit tests to test for these cases and fixed static variable storage problem which was causing a race condition when running unit tests.

### CODE REVIEW

File: transform-plugins/src/main/java/io/cdap/plugin/ValueMapper.java [hunk 1]
```diff
@@ -77,55 +82,56 @@ public Config(String mapping, String defaults) {
...
+   * It is required to use in configurePiperline as well as transform method (run in initialize).
+   */
+  private void parseConfiguration(Config config, FailureCollector collector) {
+    if (!config.defaults.isEmpty()) {
+      String[] defaultsList = config.defaults.split(",");
+      for (String defaultValue : defaultsList) {
+        String[] defaultsArray = defaultValue.split(":");
+        if (defaultsArray.length != 2) {
+          collector.addFailure(
+              String.format("Invalid default: %s.", defaultValue),
```
Comment (albertshau) [inline transform-plugins/src/main/java/io/cdap/plugin/ValueMapper.java]: indentation, here and rest of PR

File: transform-plugins/src/test/java/io/cdap/plugin/ValueMapperTest.java [hunk 2]
```diff
@@ -367,6 +385,83 @@ public void testMappingValidation() throws Exception {
...
+
+    MockPipelineConfigurer configurer = new MockPipelineConfigurer(inputSchema);
+    ValueMapper mapper = new ValueMapper(config);
+    String caughtException = "";
+    try {
+      mapper.configurePipeline(configurer);
+    } catch (Exception e) {
+      caughtException = e.getMessage();
+    }
+    Assert.assertEquals(validationExceptionMessage, caughtException);
```
Comment (albertshau) [inline transform-plugins/src/test/java/io/cdap/plugin/ValueMapperTest.java]: shouldn't check exception message

File: transform-plugins/src/main/java/io/cdap/plugin/ValueMapper.java (hunk 1 above)
Comment (dli357) [reply transform-plugins/src/main/java/io/cdap/plugin/ValueMapper.java]: Fixed

File: transform-plugins/src/test/java/io/cdap/plugin/ValueMapperTest.java (hunk 2 above)
Comment (dli357) [reply transform-plugins/src/test/java/io/cdap/plugin/ValueMapperTest.java]: Fixed

**Comment by dli357** [comment]
Overall, the validation logic in this plugin is rather complicated, but for now, due to the volume of plugins that need updating, we will simply patch it and revisit it later.

**Review by albertshau — APPROVED** [review]

**Pull request merged**